src/main/resources/templates/create_test.html

```diff
@@ -15,104 +18,102 @@
+            padding-top: 40px;
+            padding-bottom: 40px;
+            background-color: #f5f5f5;
+        }
+
+        .form-signin {
+            max-width: 330px;
+            padding: 15px;
+        }
+
+        .form-signin .form-floating:focus-within {
+            z-index: 2;
+        }
+
+        .form-signin input[type="email"] {
+            margin-bottom: -1px;
+            border-bottom-right-radius: 0;
+            border-bottom-left-radius: 0;
+        }
+
+        .form-signin input[type="password"] {
+            margin-bottom: 10px;
+            border-top-left-radius: 0;
+            border-top-right-radius: 0;
+        }
 
     </style>
 
 </head>
 <body class="text-center">
 
 
-
 <main class="form-signin w-100 m-auto">
     <form th:action="@{/home/create_test}" th:method="post" th:object="${test}">
         <h1 class="h3 mb-3 fw-normal">Input test data</h1>
 
         <div class="form-floating">
-            <input type="text" th:field="*{name}" class="form-control" id="floatingInput" placeholder="name"  />
+            <input type="text" th:field="*{name}" class="form-control" id="floatingInput" placeholder="name"/>
             <label for="floatingInput">test name</label>
 
 
         </div>
 
         <div class="form-floating">
-            <input type="text" th:field="*{duration}" class="form-control" id="floatingInput2" placeholder="duration"  />
+            <input type="text" th:field="*{duration}" class="form-control" id="floatingInput2" placeholder="duration"/>
             <label for="floatingInput2">duration</label>
         </div>
 
 
         <div class="form-floating">
-            <input type="text" th:field="*{group}" class="form-control" id="floatingInput3" placeholder="group"  />
+            <input type="text" th:field="*{group}" class="form-control" id="floatingInput3" placeholder="group"/>
             <label for="floatingInput3">group</label>
         </div>
 
-        <button class="w-100 btn btn-lg btn-primary" type="submit" >create</button>
+        <button class="w-100 btn btn-lg btn-primary" type="submit">create</button>
         <br><br>
 
 
-
     </form>
 </main>
 
 <script>
-    groupInput= document.getElementById("floatingInput3")
-    groupInput.addEventListener("keyup", function(e) {
-    if(/(([a-z]{1,})|([A-Z]{1,}))-[1-9]{2}/.test(this.value )) {
-
-
-        $.ajax({
-            //headers: {"X-CSRFToken": token},
-            url: "/subload/check_group",
-            type: 'POST',
-            data: {group: this.value},
-            dataType: "text",
-            success: function (data) {
-                if (JSON.parse(data).groupExist) {
-                    groupInput.style.backgroundColor = "green";
-                } else {
-                    groupInput.style.backgroundColor = "red";
+    groupInput = document.getElementById("floatingInput3")
+    groupInput.addEventListener("keyup", function (e) {
+        if (/(([a-z]{1,})|([A-Z]{1,}))-[1-9]{2}/.test(this.value)) {
```

Comment:
This regex is very cool but I don't intend to spend an hour trying to break it down. 
If it cannot be shortened, at least move it as a variable

Reply:
done